tests/test_clients.py

```diff
@@ -30,55 +27,70 @@ def client_token_path_override(file_config_mock, function_scope_prefix, tmp_path
     """
     Ensure each running client has a different path for the token, otherwise tests cannot run in parallel
     """
+    from rucio.common.config import config_set
     config_set('client', 'auth_token_file_path', str(tmp_path / f'{function_scope_prefix}token'))
 
 
 @pytest.mark.usefixtures("client_token_path_override")
 class TestBaseClient:
     """ To test Clients"""
 
+    from rucio.common.config import config_get
+
     cacert = config_get('test', 'cacert')
     usercert = config_get('test', 'usercert')
     userkey = config_get('test', 'userkey')
 
     def testUserpass(self, vo):
         """ CLIENTS (BASECLIENT): authenticate with userpass."""
         creds = {'username': 'ddmlab', 'password': 'secret'}
+        from rucio.client.baseclient import BaseClient
         client = BaseClient(account='root', ca_cert=self.cacert, auth_type='userpass', creds=creds, vo=vo)
         print(client)
 
     def testUserpassWrongCreds(self, vo):
         """ CLIENTS (BASECLIENT): try to authenticate with wrong username."""
         creds = {'username': 'wrong', 'password': 'secret'}
+        from rucio.client.baseclient import BaseClient
+
         with pytest.raises(CannotAuthenticate):
             BaseClient(account='root', ca_cert=self.cacert, auth_type='userpass', creds=creds, vo=vo)
 
     def testUserpassNoCACert(self, vo):
         """ CLIENTS (BASECLIENT): authenticate with userpass without ca cert."""
         creds = {'username': 'wrong', 'password': 'secret'}
+        from rucio.client.baseclient import BaseClient
+
         with pytest.raises(CannotAuthenticate):
             BaseClient(account='root', auth_type='userpass', creds=creds, vo=vo)
 
     def testx509(self, vo):
         """ CLIENTS (BASECLIENT): authenticate with x509."""
+        from rucio.client.baseclient import BaseClient
+
         creds = {'client_cert': self.usercert,
                  'client_key': self.userkey}
         BaseClient(account='root', ca_cert=self.cacert, auth_type='x509', creds=creds, vo=vo)
 
     def testx509NonExistingCert(self, vo):
         """ CLIENTS (BASECLIENT): authenticate with x509 with missing certificate."""
         creds = {'client_cert': '/opt/rucio/etc/web/notthere.crt'}
+        from rucio.client.baseclient import BaseClient
```

Comment:
why is `BaseClient` imported separately for each test?

Reply:
When BaseClient is imported it runs `setup_logger`, which uses `config_get`, and thus, initializes the config, and renders the added test pointless. I can move the new tests to a separate file to avoid the separate imports, but I figured the added test belongs here.